### ISSUE TITLE
keybinding to make new workspace independent of mod+1,2,3,4 etc.

### ISSUE THREAD

**Issue by ghost** [Airblader/i3#427]
TLDR: a keybinding to create new workspaces independent of numbered workspaces, that can reorder themselves according to numeric value if a workspace between the min/max numeric value is destroyed.

Hi, I would like to make a feature request. Before using i3-gaps, I was using bspwm as a result I have become used to using persistant workspaces, I had a look online to see how to enable this 'feature' and found this post https://github.com/i3/i3/issues/1902

Firstly, I would like to say this is not another request for persistant workspaces! Having read through the above link, I can understand and sympathise with the i3 philosphy regarding workspaces :-)

However, in the above link Airblader (who I assume will be reading this, so hello there) made a valid point about use case scenarios, here is the quote:

"I would argue that in general users take one of two approaches to workspaces:

- They dedicate specific workspaces to specific programs (which is my personal approach). In this case it quickly becomes muscle memory to know which number to choose and there is no more thinking involved.

- They use workspaces freely as they go. In this case it doesn't really matter which workspace they choose anyway."

Personally, I am in the second use case scenario. I don't care about keeping track of which workspace name/number I am on and therefore it doesn't matter to me which workspace I choose. I am a fairly causual linux user, in as much as I may only use 2-4 workspaces at a time.

I believe the request for persistant workspaces is the result of seeing things like various polybar setups (I also use polybar) and seeing cool little icons to show all the workspaces available. On a window manager like bspwm with polybar added, in my case at least, if I use a maximum of 4 workspaces, I can visually determine I have 4 workspaces as they are all visable regardless of focus. I can determine I am on worksapce 2 for example by configuring a focused workspace icon in polybar.

The issue I have is that if I use a polybar setup similar with i3 and configure polybar with a simple icom to show focused/unfocused workspaces. For example a solid circle icon to show this is the workspace I am on and an outline of a circle to show an open but unfocused workspace. It can only work for as long as I remember which workspace I am on... If I start my session on worksapce 1 and use a container and then open a container on workspace 2 and close the container on workspace 1, Looking at polybar with my current setup could lead me to believe workspace 2 is workspace 1 because I only have one workspace displayed in the bar with a circle icon in. Later on when I press mod+2 nothing happens becauee I am already on workspace 2.

Now I fully understand and appreciate that I could save myself the inconvenience and just show which number workspace I am on with polybar... or find different icons to represent numeric values.

However, having thought about it last night, and not actually having an issue with the workspaces destructing when not being used, I tried to think of a solution that might still fit in with the i3 philosophy, yet give people like me that don't care about workspace names/numbers or just want some pretty icons in their bar a workable option.

And this quote got me thinking, "They use workspaces freely as they go. In this case it doesn't really matter which workspace they choose anyway." because it is an accepted way that some of your users use i3.

I am not a coder or anything like that so I do not know how hard or easy this would be to implement, nor whether it is feasible or desired by the community.

Simply put, an option to create new workspaces on the fly that can still be destroyed when not in use but bind this to a specific key (mod+<key>.) The key combo (mod+<key>) could create as many new workspaces as possible or be limited to a max number specified by the user but numeric vales assigned to the workspaces would always be in order.

Example:
- user logs in on workspace 1 opens container and wants a new container in a new workspace (user does not care what this workspace is called) the user presses mod+<key> to create new workspace. i3 calls it workspace 2 because it is the second workspace visable in the bar.
- user needs an additional workspace, user presses mod+<key> for the new workspace. i3 calls it workspace 3 again because it is the third visable workspace in the bar.
- user is finished with workspace 2, closes the container and jumps back to workspace 1. Workspace 2 is destroyed and workspace 3 is now renamed as workspace 2.
- if a new workspace was summoned from workspace 1 or 'workspace 2' (which was orginally workspace 3), the new workspace is called workspace 3.

This would avoid having a workspace 1, workspace 4 and workspace 7 for example, and would probably be best suited for those users, like me that don't care about the names/numbers f their activr workspaces, but just want a new workspace.

This would allow easier navigation with the mod key + number of the workspace, for those wanting simple icons to display workspaces in their bar of choice and most importantly still fit in with i3 philosophy regarding destroying unused workspaces. If you see 3 icons you have workspaces 1, 2 and 3.

Sorry for the essay, I just wanted to be as detailed as possible I hope my suggestion makes sense. If such a feature is already available please forgive my ignorance I search online last night returned no useful links for me to implent it.

Thanks, Rob
